docs/rules/no-extraneous-import.md

```diff
@@ -26,57 +26,18 @@ This rule warns `import` declarations of extraneous modules.
 
 #### allowModules
 
-Some platforms have additional embedded modules.
-For example, Electron has `electron` module.
-
-We can specify additional embedded modules with this option.
-This option is an array of strings as module names.
-
-```json
-{
-    "rules": {
-        "n/no-extraneous-import": ["error", {
-            "allowModules": ["electron"]
-        }]
-    }
-}
-```
+This can be configured in the rule options or as a shared setting [`settings.allowModules`](../shared-settings.md#allowmodules).
+Please see the shared settings documentation for more information.
 
 #### resolvePaths
 
-Adds additional paths to try for when resolving imports.
-If a path is relative, it will be resolved from CWD.
-
-Default is `[]`
+This can be configured in the rule options or as a shared setting [`settings.resolvePaths`](../shared-settings.md#resolvepaths).
+Please see the shared settings documentation for more information.
 
 #### convertPath
 
-- `exclude`: TODO
-- `include`: TODO
-- `replace`: TODO
```

Comment:
@bmish it causes `eslint-docs-generator` to report an error, when moving the options docs to another page(shared-settings). Is there a way to ignore the error?

I'm glad to help making a PR in `eslint-docs-generagor` if you want. 😄

Reply:
Since it looks like the rule options are all mentioned in a shared doc instead, it should be safe for you to use the option `--rule-doc-section-options false` to disable the check (https://github.com/bmish/eslint-doc-generator#configuration-options).

Reply:
The issue is that we mention the root option, and reference it, eg in `no-extraneous-import`:

We reference `convertPath` [docs/rules/no-extraneous-import.md#convertpath](https://github.com/scagood/eslint-plugin-n/blob/shared-settings/docs/rules/no-extraneous-import.md#convertpath) and point to the shared setting [docs/shared-settings.md#convertpath](https://github.com/scagood/eslint-plugin-n/blob/shared-settings/docs/shared-settings.md#convertpath)

The schema is defined here: [lib/rules/no-extraneous-import.js#L27-L31](https://github.com/scagood/eslint-plugin-n/blob/shared-settings/lib/rules/no-extraneous-import.js#L27-L31)

The issue is that the missing options are child properties of `convertPath`:
- `convertPath.include`
- `convertPath.exclude`
- `convertPath.replace`

```
`no-extraneous-import` rule doc should have included rule option: include
`no-extraneous-import` rule doc should have included rule option: exclude
`no-extraneous-import` rule doc should have included rule option: replace
```

Reply:
The existing eslint-doc-generator option `--rule-doc-section-options` is all-or-nothing in terms of checking for the options section and any named options it can detect. So with your planned design, you may just have to disable the option. Let me know if you have ideas for more granular options to control these checks.

Reply:
wonderful job! The option looks useful! 👍

Reply:
🤔 I am not sure I want to use that option, I don't have much time today, I will try to come up with something tomorrow!

Reply:
Another option is to extend `eslint-doc-generator` to ignore the specific rules - Either way works for me.

Reply:
@bmish Would something like this work for you?
<img width="1027" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/e077ca95-5a6b-4097-902e-f8cb4bf23f65">

Or if you don't like the dot separators something like this?
<img width="887" alt="image" src="https://github.com/eslint-community/eslint-plugin-n/assets/2230835/7a854283-e671-4cfc-940d-ecc12e92de29">

Reply:
Effectivly its an option to check to see if you only want to check the parent option (`convertPath`) or the child option (`convertPath`) 🤔

I was thinking that you could have it similar to a `depth` option.

eg:
| Option Path | depth |
| --- | --- |
| `convertPath` | 1 |
| `convertPath.include` | 2 |
| `convertPath.include.something` | 3 |

Then the option could be something like `--rule-doc-options-depth=1`.

I am open to contributing if this is something you like the sound of, it obviously need a little bit more planning than a quick couple of edits in a node_modules transpiled file 😂

Reply:
We can turn off this option for now and use that option when it was supported in `eslint-doc-generator`.
wdyt?

Reply:
mmm, I don't love the option, as you'd need to be more vigilant in PRs.

If that is fine with you, then we can proceed with the option.

Reply:
I don't love it, but its disabled here:

[`2f27eb8` (#115)](https://github.com/eslint-community/eslint-plugin-n/pull/115/commits/2f27eb867cb72fd4d88603af105341c420f00773)

Reply:
to avoid breaking changes: 
* `--rule-doc-section-options=true` can be treated as `depth: Infinity`
* `--rule-doc-section-options=false` as `depth: 0`